tests/beaker_tests/cisco_bgp/test_bgp.rb

```diff
@@ -0,0 +1,260 @@
+    bestpath_compare_routerid:       'true',
+    bestpath_cost_community_ignore:  'true',
+    bestpath_med_confed:             'true',
+    bestpath_med_missing_as_worst:   'true',
+    bestpath_med_non_deterministic:  'true',
+  },
+}
+
+tests[:non_def_3] = {
+  desc:           '2.3 Non Defaults Part 3',
+  title_pattern:  "#{asn} default",
+  manifest_props: {
+    cluster_id:                             '10.0.0.1',
+    confederation_id:                       '99',
+    confederation_peers:                    '55 23.4 88 200.1',
+    enforce_first_as:                       'true',
+    fast_external_fallover:                 'true',
+    flush_routes:                           'false',
+    graceful_restart:                       'true',
+    graceful_restart_helper:                'true',
+    graceful_restart_timers_restart:        '130',
+    graceful_restart_timers_stalepath_time: '310',
+  },
+}
+
+tests[:non_def_4] = {
+  desc:           '2.4 Non Defaults Part 4',
+  title_pattern:  "#{asn} default",
+  manifest_props: {
+    isolate:                     'false',
+    log_neighbor_changes:        'true',
+    maxas_limit:                 '50',
+    router_id:                   '192.168.0.66',
+    shutdown:                    'true',
+    suppress_fib_pending:        'true',
+    timer_bestpath_limit:        '255',
+    timer_bestpath_limit_always: 'true',
+    timer_bgp_holdtime:          '110',
+    timer_bgp_keepalive:         '45',
+  },
+}
+
+# Platform-specific tests
+tests[:non_def_plat_1] = {
+  desc:           '3.1 Default Properties Platform-specific Part 1',
+  platform:       'n(3|9)k',
+  title_pattern:  "#{asn} default",
+  manifest_props: {
+    neighbor_down_fib_accelerate: 'true'
+  },
+}
+
+tests[:title_patterns] = {
+  preclean:       'cisco_bgp',
+  manifest_props: {},
+  resource:       { 'ensure' => 'present' },
+}
+
+# Title Pattern Test Hash
+titles = {}
+titles['T.1'] = {
+  title_pattern: '2',
+  title_params:  { vrf: 'blue' },
+}
+
+# This helper tests a test case in vrf context. This allows for testing a vrf
+# while an existing config is present in vrf default.
+def test_harness_bgp_vrf(tests, id, vrf)
+  orig_desc = tests[id][:desc]
+  tests[id][:desc] += " (vrf #{vrf})"
+  tests[id][:title_pattern] = "#{tests[:asn]} #{vrf}"
+
+  test_harness_run(tests, id)
+  tests[id][:desc] = orig_desc
+end
+
+#################################################################
+# TEST CASE EXECUTION
+#################################################################
+test_name "TestCase :: #{testheader}" do
+  # -------------------------------------------------------------------
+  logger.info("\n#{'-' * 60}\nSection 1. Default Property Testing")
+
+  # -----------------------------------
+  test_harness_run(tests, :default)
+  test_harness_run(tests, :default_plat_1)
+
+  id = :default
+  tests[id][:ensure] = :absent
+  tests[id].delete(:preclean)
+  test_harness_run(tests, id)
```

Comment:
Why are you running the `:default` testcase twice?  Once on line 221 and again on line 227?